airbyte-integrations/connectors/source-chargebee/source_chargebee/streams.py

```diff
@@ -0,0 +1,172 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from datetime import datetime, timezone
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional
+
+import backoff
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams import Stream
+from chargebee.api_error import OperationFailedError
+from chargebee.list_result import ListResult
+from chargebee.model import Model
+from chargebee.models import Addon, Customer, Invoice, Order, Plan, Subscription
+
+# Backoff params below
+# according to Chargebee's guidance on rate limit
+# https://apidocs.chargebee.com/docs/api?prod_cat_ver=2#api_rate_limits
+MAX_TRIES = 10  # arbitrary max_tries
+MAX_TIME = 90  # because Chargebee API enforce a per-minute limit
+
+
+class ChargebeeStream(Stream):
+    primary_key = "id"
+    cursor_field = "updated_at"
+    page_size = 100
+    api: Model = None
+
+    def __init__(self, start_date: str):
+        # Convert `start_date` to timestamp(UTC).
+        self._start_date = int(datetime.strptime(start_date, "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc).timestamp())
```

Comment:
I understand that all state data will be `timestamp` and, accordingly, have the `integer` type, but perhaps it will be more convenient to use `Pendulum` here?

Reply:
Done.